jax/_src/numpy/polynomial.py

```diff
@@ -85,19 +85,20 @@ def _roots_with_zeros(p: Array, num_leading_zeros: int) -> Array:
 """)
 def roots(p: ArrayLike, *, strip_zeros: bool = True) -> Array:
   check_arraylike("roots", p)
-  p_arr = atleast_1d(*promote_dtypes_inexact(p))
+  p_arr = atleast_1d(promote_dtypes_inexact(p)[0])
```

Comment:
The list has length 1.  There are various ways to make this clear.  It could be moved on a different line and unpacked using
```
promoted_p, = promote_dtypes_inexact(p)
```
Or you could use more_itertools.one.

Having the type checker know that there is exactly one element allows it to realize that `p_arr` is an array rather than a list of arrays.